src/cmd/run.js

```diff
@@ -38,14 +39,16 @@ export async function run (componentPath, args) {
       }
       catch {}
       let len = preview2ShimPath.length;
-      preview2ShimPath = resolve(preview2ShimPath, '../../../node_modules/@bytecodealliance/preview2-shim');
+      preview2ShimPath = resolve(preview2ShimPath, '..', '..', '..', 'node_modules', '@bytecodealliance', 'preview2-shim');
       if (preview2ShimPath.length === len) {
         console.error(c`{red ERR}: Unable to locate the {bold @bytecodealliance/preview2-shim} package, make sure it is installed.`);
-        return;
+        process.exitCode = 1;
```

Comment:
This should still immediately return.

Reply:
I started using `process.exitCode` as a substitute for `process.exit()`, to ensure that the `finally` block would still execute. But that obviously hasn't worked in this case.

I would like to see errors handled more consistently: currently, they can cause (some combination of) a return, an exit code to be set, or an exception to be thrown. This will become particularly important if `run()` is added to the API (which I would like to see eventually).